Source/EasyNetQ/Persistent/PersistentChannel.cs

```diff
@@ -46,13 +47,71 @@ IEventBus eventBus
     }
 
     /// <inheritdoc />
-    public async Task<TResult> InvokeChannelActionAsync<TResult, TChannelAction>(
+    public ValueTask<TResult> InvokeChannelActionAsync<TResult, TChannelAction>(
         TChannelAction channelAction, CancellationToken cancellationToken = default
     ) where TChannelAction : struct, IPersistentChannelAction<TResult>
     {
         if (disposed)
             throw new ObjectDisposedException(nameof(PersistentChannel));
 
+        cancellationToken.ThrowIfCancellationRequested();
+
+        return TryInvokeChannelActionFast<TResult, TChannelAction>(channelAction, out var result)
+            ? new ValueTask<TResult>(result)
+            : new ValueTask<TResult>(InvokeChannelActionSlowAsync<TResult, TChannelAction>(channelAction, cancellationToken));
+    }
+
+    /// <inheritdoc />
+    public void Dispose()
+    {
+        if (disposed)
+            return;
+
+        disposed = true;
+        disposeCts.Cancel();
+        mutex.Dispose();
+        CloseChannel();
+        disposeCts.Dispose();
+    }
+
+    private bool TryInvokeChannelActionFast<TResult, TChannelAction>(
+        in TChannelAction channelAction, [MaybeNullWhen(false)] out TResult result
+    ) where TChannelAction : struct, IPersistentChannelAction<TResult>
+    {
+        if (mutex.TryAcquireImmediately(out var releaser))
+        {
+            try
+            {
+                var channel = initializedChannel ??= CreateChannel();
+                // ReSharper disable once PossiblyImpureMethodCallOnReadonlyVariable
+                result = channelAction.Invoke(channel);
+                return true;
+            }
+            catch (Exception exception)
+            {
+                var exceptionVerdict = GetExceptionVerdict(exception);
+                if (exceptionVerdict.CloseChannel)
+                    CloseChannel();
+
+                if (exceptionVerdict.Rethrow)
+                    throw;
+
+                logger.Error(exception, "Failed to invoke channel action, invocation will be retried");
```

Comment:
```suggestion
                logger.Error(exception, "Failed to invoke channel action on fast path, invocation will be retried on slow path");
```

something like that to distinguish `TryInvokeChannelActionFast` from `InvokeChannelActionSlowAsync`.

Reply:
Done